### DISCUSSION TITLE
Python version assert fails on odd numbers

### DISCUSSION THREAD

**Discussion by martin-niklasson** [PlummersSoftwareLLC/Primes#231] [Improvement suggestions]
Try:
`sieve = prime_sieve(999999)                        #  Calc the primes up to a million-1`

Result:
`    assert(count == this.countPrimes())
AssertionError
`

The reason is that keyword int in Python doesn't do what a C++ programmer thinks:
`this.rawbits = [True] * (int((this.sieveSize+1)/2))`

Fix by:
`from math import sqrt,floor                       # Used by the sieve`
and
`this.rawbits = [True] * (floor((this.sieveSize)/2))`

**Comment by hegdenischay** [comment]
Why not just `this.rawbits = [True] * ((this.sieveSize)//2))`?
I thought it'd be more performant, since it's an operator rather than a function, but I don't see any real performance benefit on my own machine.

**Comment by davepl** [comment]
Check in a fix, I'll gladly accept!
